### PR TITLE
Cult Stun & Blood Rites Changes

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Cultist Stun Hand's power is reduced by 33% when the cult has Risen (red eyes), and by 66% when it has Ascended (halos).

Stunned non-cultists recieve dizziness and temporary eye damage, the duration increasing the more the cult grows.

Effectively, while stuns are weakened, the stun effect applies dizziness as well now, helping against stun-resistant targets.

Stacking multiple spells of one type is no longer allowed. (Oranges wanted this)

Blood Rites no longer disappear at 0 charges.

## Why It's Good For The Game

> Cultist Stun Hand's power is reduced by 33% when the cult has Risen (red eyes), and by 66% when it has Ascended (halos).

This spell is infamous for being a 2018-era stun. Nerfing it completely, however, would be a large issue for early cult as it'd make it extremely hard for them to convert members. Thus, the power is weakened the more the cult grows.

> Stunned non-cultists recieve dizziness and temporary eye damage, the duration increasing the more the cult grows.

This serves two functions: While the non-cultist manages to get up earlier, they're still stumbling around confused so it's not like they can take advantage of it to bring down cultists too well. Additionally, it makes stun immune targets still suffer some form of negative effect from the spell, making hulks less of a game-over for cultists.

> Stacking multiple spells of one type is no longer allowed. (Oranges wanted this)

Orange man bad0

> Blood Rites no longer disappear at 0 charges.

This was annoying.

## Changelog

:cl:
balance: Cultist Stun Hand's power is reduced by 33% when the cult has Risen (red eyes), and by 66% when it has Ascended (halos).
balance: Stunned non-cultists recieve dizziness and temporary eye damage, the duration increasing the more the cult grows.
balance: Effectively, while stuns are weakened, the stun effect applies dizziness as well now, helping against stun-resistant targets.
balance: Stacking multiple spells of one type is no longer allowed. (Oranges wanted this)
qol: Blood Rites no longer disappear at 0 charges.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/cult/blood_magic.dm
Comment (necromanceranne) [inline code/modules/antagonists/cult/blood_magic.dm:110]: I don't think you've applied this new var to blood rites.

Comment (carlarctg) [reply code/modules/antagonists/cult/blood_magic.dm:110]: oh yeah i tested cult stun but forgor to test blood rites

**Comment by CCC23** [comment]
thank god no more 18 year stun hand

**Comment by anclandor** [comment]
I feel like not being able to have more than one of a spell really takes away from the need to think about what spells you are taking.  Do you give up a slot for blood rites or emp?  Or put them all into the stunhand?  But on the other hand it might lead to some more use of the hide/reveal spell and other spells that get slept on.

**Comment by necromanceranne** [comment]
It may be worth considering adding blood rites as a permanent spell, and blood charges wiping on holy water instead of the spell itself. Mostly because the spell is _technically_ permanent with this change. If cultists have to lose a max spell to make this happen, so be it.

**Comment by carlarctg** [comment]
> It may be worth considering adding blood rites as a permanent spell, and blood charges wiping on holy water instead of the spell itself. Mostly because the spell is _technically_ permanent with this change. If cultists have to lose a max spell to make this happen, so be it.

That is a really cool idea and gets around the new power limit this pr adds

**Comment by Livrah** [comment]
Will the new converts still be under dizziness and eye damage effects?

**Comment by DATA-xPUNGED** [comment]
Does the decrease of stun hand power just makes all times shorter? I think it'd be interesting if on the halo step, it turned into a knockdown like the heretic hand

**Comment by AtomTheProphet** [comment]
My only issue with this is that we can't have multiple spells of the same type. Otherwise though, it's really good!

**Comment by necromanceranne** [comment]
I guess giving it a bit more thought, there is a few things I would bring up.

How much of a meaningful differences does current stun hand _actually_ make on later cult? More than a few rounds I've seen stun hand see a lot less use than just cutting down opposition with brute force and numbers. One stun hand might stop one opponent, but it's a one-and-done spell, you'd have to reprepare it to use it again. But just bulldozing opponents with your swords and halberds is often something that is more sustainable across multiple fights with the crew. I'd imagine this change would push knowledgeable players into that direction, is that the expected transition for cult?

On the topic of knowledgeable players, doesn't this add another layer of knowledge checking for cultists? How would they know that this spell is weaker once the cult starts to gain numbers, if they had been using the spell up to the point they have hit stage two or three? This step is already one of the more difficult processes for cultists to understand already.

I can't disagree that the cult stun is very outdated, but so is the mode as a whole. And if the intent is to shift the cult away from those early capture stuns into skirmishing, is this the most elegant method? Would it maybe be better to outright remove access to the spell at a certain stage in the cults progression, rather than undercut player expectations?

**Comment by carlarctg** [comment]
> I guess giving it a bit more thought, there is a few things I would bring up.
> 
> How much of a meaningful differences does current stun hand _actually_ make on later cult? More than a few rounds I've seen stun hand see a lot less use than just cutting down opposition with brute force and numbers. One stun hand might stop one opponent, but it's a one-and-done spell, you'd have to reprepare it to use it again. But just bulldozing opponents with your swords and halberds is often something that is more sustainable across multiple fights with the crew. I'd imagine this change would push knowledgeable players into that direction, is that the expected transition for cult?
> 
> On the topic of knowledgeable players, doesn't this add another layer of knowledge checking for cultists? How would they know that this spell is weaker once the cult starts to gain numbers, if they had been using the spell up to the point they have hit stage two or three? This step is already one of the more difficult processes for cultists to understand already.
> 
> I can't disagree that the cult stun is very outdated, but so is the mode as a whole. And if the intent is to shift the cult away from those early capture stuns into skirmishing, is this the most elegant method? Would it maybe be better to outright remove access to the spell at a certain stage in the cults progression, rather than undercut player expectations?

I agree, I simply got tired of hearing countless arguments on the discord and lazily threw this together as 'my lazy take on the cult hand debate'



> Will the new converts still be under dizziness and eye damage effects?

Good point, me fix

**Comment by DaydreamIQ** [comment]
Are heretics / magic immune people affected by the dizziness and eye damage? The whole point of heretics being unaffected was so the cult couldn't get an ez BBS after all.

**Comment by necromanceranne** [comment]
> Are heretics / magic immune people affected by the dizziness and eye damage? The whole point of heretics being unaffected was so the cult couldn't get an ez BBS after all.

I'm not sure why you would assume it would given it only changed the effects of the spell and nothing else. Those checks return the entire proc early, so they're before any affects of the spell are applied.

**Comment by SimplySmiley123** [comment]
> I guess giving it a bit more thought, there is a few things I would bring up.
> 
> How much of a meaningful differences does current stun hand _actually_ make on later cult? More than a few rounds I've seen stun hand see a lot less use than just cutting down opposition with brute force and numbers. One stun hand might stop one opponent, but it's a one-and-done spell, you'd have to reprepare it to use it again. But just bulldozing opponents with your swords and halberds is often something that is more sustainable across multiple fights with the crew. I'd imagine this change would push knowledgeable players into that direction, is that the expected transition for cult?
> 
> On the topic of knowledgeable players, doesn't this add another layer of knowledge checking for cultists? How would they know that this spell is weaker once the cult starts to gain numbers, if they had been using the spell up to the point they have hit stage two or three? This step is already one of the more difficult processes for cultists to understand already.
> 
> I can't disagree that the cult stun is very outdated, but so is the mode as a whole. And if the intent is to shift the cult away from those early capture stuns into skirmishing, is this the most elegant method? Would it maybe be better to outright remove access to the spell at a certain stage in the cults progression, rather than undercut player expectations?

It's not used 24/7 all throughout the round, but it is effectively a one click win against any and all opponents bar chaplains with their nullrod on them or someone carrying a holy melon in their hands. But those are usually lone targets you cna bulldoze through anyway later on.
It also effectively makes mindshields pointless to give out on cult rounds as it only makes it worse for the crew-side, as they still will get stun handed but now will be round removed and turned into a construct.

File: code/modules/antagonists/cult/blood_magic.dm
Comment (Arturlang) [inline code/modules/antagonists/cult/blood_magic.dm:430]: This should be a bubble message, people won't see it in combat, consider also adding the fact that it weakens in the description

**Comment by HerCoyote23** [comment]
I feel like a decent alternative to making cult hand arbitrarily less good over time while keeping it wacko OP early on might be to make it a multi-hit affair.
Maybe it does 50 stam damage, silences, and disorients people on hit, but doesn't expend itself on hit, instead acting more like Chap's God Hand. Sort of like a baton that silences and has unlimited charge.
Then you can still pretty easily kidnap someone without being called out, but there's no more 2018-hardstun salt.

**Comment by wesoda25** [comment]
> I feel like a decent alternative to making cult hand arbitrarily less good over time while keeping it wacko OP early on might be to make it a multi-hit affair. Maybe it does 50 stam damage, silences, and disorients people on hit, but doesn't expend itself on hit, instead acting more like Chap's God Hand. Sort of like a baton that silences and has unlimited charge. Then you can still pretty easily kidnap someone without being called out, but there's no more 2018-hardstun salt.

I'd say merge this and then eventually try that once people are less reliant on cult stuns (which is bound to happen due to this making them less reliable). As is cult is simply too hard and I think the stun is a necessary crutch for early game, specifically for inexperienced players. For that reason there's not really any sort of comprehensive solution to the stun besides regearing cult to be less of a TDM or making it even more similar to revs (which would be a shame, I think).

File: code/modules/antagonists/cult/runes.dm
Comment (MrMelbert) [inline code/modules/antagonists/cult/runes.dm:293]: `AdjustAllImmobility(0)` = "Reduce all stuns by 0 seconds". You want `SetAllImmobility`

File: code/modules/antagonists/cult/runes.dm
Comment (MrMelbert) [inline code/modules/antagonists/cult/runes.dm:296]: Cringe comment

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/cult/runes.dm
Comment (necromanceranne) [reply code/modules/antagonists/cult/runes.dm:296]: co-authored by onule

**Comment by JohnFulpWillard** [comment]
i agree with melbert's second review but also please handle them

**Pull request closed without merging**
